Dart/src/com/jetbrains/lang/dart/Dart.bnf

```diff
@@ -313,19 +313,19 @@ private functionNative ::= 'native' (stringLiteralExpression ';' | ';' | stringL
 
 methodDeclaration ::= metadata* ('external' | 'static' | 'const')* methodDeclarationPrivate initializers? (';' | functionBodyOrNative | redirection)?
 {pin=3 mixin="com.jetbrains.lang.dart.psi.impl.AbstractDartMethodDeclarationImpl" implements="com.jetbrains.lang.dart.psi.DartComponent"}
-private methodDeclarationPrivate ::= returnType <<methodNameWrapper>> typeParameters? formalParameterList | !untypedFunctionType <<methodNameWrapper>> typeParameters? formalParameterList // todo remove, use functionSignature as in spec
+private methodDeclarationPrivate ::= returnType (<<methodNameWrapper>> | 'new') typeParameters? formalParameterList | !untypedFunctionType (<<methodNameWrapper>> | 'new') typeParameters? formalParameterList // todo remove, use functionSignature as in spec
```

Comment:
It's suspicious that here's no '.' before 'new'. Are you sure this change is needed? Tests don't fail if I revert changes in this line.

Reply:
You are correct, the syntax for named constructors kicks in making this change unnecessary, reverting here.